qaequilibrae/modules/processing_provider/exportMatrix.py

```diff
@@ -0,0 +1,64 @@
+__author__ = 'Arthur Evrard'
+
+from qgis.core import QgsProcessingAlgorithm, QgsProcessingMultiStepFeedback
+from qgis.core import QgsProcessingParameterFile, QgsProcessingParameterEnum
+from .translatableAlgo import TranslatableAlgorithm
+import importlib.util as iutil
+import os, tempfile, sys
+from pathlib import Path
+
+class exportMatrix(TranslatableAlgorithm):
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterFile('srcFile', self.tr('Matrix file to convert (.omx or .aem)'), behavior=QgsProcessingParameterFile.File, fileFilter=self.tr('AequilibraE matrix formats (*.omx *.aem)'), defaultValue=None))
+        self.addParameter(QgsProcessingParameterFile('destFolder', self.tr('Export matrices to'), behavior=QgsProcessingParameterFile.Folder, defaultValue=None))
+        self.addParameter(QgsProcessingParameterEnum('outputformat', self.tr('Format to use for export'), options=['.csv','.omx','.aem'], defaultValue='.csv'))
+
+    def processAlgorithm(self, parameters, context, model_feedback):
+        results = {}
+        outputs = {}
+
+        pathSource=parameters['srcFile']
+        fileformat=['.csv','.omx','.aem']
+        pathDest=os.path.join(parameters['destFolder'], Path(pathSource).stem+fileformat[parameters['outputformat']])
+
+        # Checks if we have access to aequilibrae library
+        has_aeq = iutil.find_spec("aequilibrae") is not None
+
+        if not has_aeq:
+            sys.exit(self.tr('AequilibraE library not found'))
```

Comment:
can we extract this into a helper method?
```suggestion

        def ensure_aeq_available():
            if iutil.find_spec("aequilibrae") is None:
                sys.exit(self.tr('AequilibraE library not found'))
```

and then just call that method in all these providers

Reply:
Good idea @jamiecook, will do that once we get a solution for #256 which prevents me from doing any more checking at the moment